spyder/plugins/ipythonconsole/widgets/debugging.py

```diff
@@ -100,11 +108,26 @@ def callback(line):
                 self.kernel_client.input(code)
             else:
                 self.kernel_client.input(line)
+
+            self._highlighter.highlighting_on = False
         if self._reading:
             self._reading = False
-        self._readline(msg['content']['prompt'], callback=callback,
+
+        prompt = msg['content']['prompt']
+
+        self._highlighter.highlighting_on = True
+        self._readline(prompt, callback=callback,
                        password=msg['content']['password'])
 
+    def _show_prompt(self, prompt=None, html=False, newline=True):
+        """
+        Writes a new prompt at the end of the buffer.
+        """
+        if prompt in ['(Pdb) ', 'ipdb> ']:
+            html = True
+            prompt = '<span class="in-prompt">%s</span>' % prompt
+        super(DebuggingWidget, self)._show_prompt(prompt, html, newline)
```

Comment:
Does this addition interfere with users trying to customize prompts?

This can be done in

    Preferences > IPython Console > Advanced settings > Prompts

Reply:
Not that I could see. It just marks '(Pdb) ' and  'ipdb> ' with class="in-prompt"

Reply:
basically adds HTML around them

Reply:
This is the same HTML used arounf `In [1]`

Reply:
Ok, thanks for taking a look at this!